### PR TITLE
Update k8s.io dependencies to v0.20.6

### PR DESCRIPTION
**Pull request by akrejcir**
**What this PR does / why we need it**:
Updated k8s.io dependencies to `v0.20.6`.

**Release note**:
```release-note
None
```

### CODE REVIEW

File: go.mod [hunk 1]
```diff
@@ -16,9 +16,9 @@ require (
 	github.com/pkg/errors v0.9.1
 	github.com/spf13/cobra v1.1.1
 	gomodules.xyz/jsonpatch/v2 v2.1.0
-	k8s.io/api v0.20.2
-	k8s.io/apiextensions-apiserver v0.20.1
-	k8s.io/apimachinery v0.20.2
+	k8s.io/api v0.20.6
```
Comment (vatsalparekh) [inline go.mod:19]: This is already being covered as part of #134

Comment (akrejcir) [reply go.mod:19]: That PR only updates to `v0.20.4`. If you change it to `v0.20.6`, I can close this PR.

Comment (vatsalparekh) [reply go.mod:19]: In the latest change, I'm updating it to `v0.20.6`

Comment (akrejcir) [reply go.mod:19]: I will not close this. We will probably need to backport this PR to `release-v0.1` branch, because it contains a security fix.

**Pull request merged**